osu.Game/Skinning/RulesetSkinProvidingContainer.cs

```diff
@@ -42,13 +45,21 @@ public RulesetSkinProvidingContainer(Ruleset ruleset, IBeatmap beatmap, [CanBeNu
             };
         }
 
+        private SkinManager skinManager;
         private ISkinSource parentSource;
 
+        private ResourceStoreBackedSkin rulesetResourcesSkin;
+
         protected override IReadOnlyDependencyContainer CreateChildDependencies(IReadOnlyDependencyContainer parent)
         {
+            skinManager = parent.Get<SkinManager>();
```

Comment:
Remove all this and just use `is DefaultSkin` or similar instead. A reference to `SkinManager` should not be required.

Reply:
Hmm, that means if one makes a modified version of `DefaultSkin`, it will still be overridden by ruleset resources, you think it's fine?

To me, it sounds like the skins shouldn't be instantiated by `Default` classes, but not sure if there's a correct other way for it.

Reply:
Is that how it currently works?

Reply:
I believe so, if one modifies `DefaultSkin` or `DefaultLegacySkin`, the `InstantiationInfo` of their modified skins point to the default classes.

Reply:
Even so, can't you just find the last matching instance in the sources list?

Reply:
Right, can do that.